### PR TITLE
Housekeeping

### PR DESCRIPTION
**Pull request by paulstraw**
A handful of changes:

* Fix typo in spec name and "migrations" content type description
* Bump contentful-management.rb from very outdated 2.6 to 3.4.0
* Fix automatic creation of "migrations" content type (ENG-2421)
* Add VCR to fix failing specs

Leaving this as draft so it's not automatically merged, but I do think it's ready for review. I want to play with this in the actual Cometeer app tomorrow before merging.

### CODE REVIEW

File: lib/contentful_migrations/migration_content_type.rb [hunk 1]
```diff
@@ -34,7 +34,7 @@ def build_migration_content_type
       content_type = space.content_types.create(
         name: migration_content_type_name,
         id: migration_content_type_name,
-        description: 'Migration Table for interal use only, do not delete'
+        description: 'Migration Table for internal use only, do not delete'
```
Comment (cpfergus1) [inline lib/contentful_migrations/migration_content_type.rb:34]: 💯

File: lib/contentful_migrations/migration_content_type.rb [hunk 2]
```diff
@@ -22,12 +22,9 @@ def resolve
   private
 
     def find_or_create_migration_content_type
-      content_type = space.content_types.find(migration_content_type_name)
-      if content_type.nil? || content_type.is_a?(Contentful::Management::Error)
-        build_migration_content_type
-      else
-        content_type
-      end
+      space.content_types.find('migrations')
```
Comment (cpfergus1) [inline lib/contentful_migrations/migration_content_type.rb:25]: Wouldn't this return nil if 'migrations' was not found and no error was thrown? 
Would a ternary help here? 
```
migrations = space.content_types.find('migrations')
migrations.nil? ? build_migration_content_type : migrations
```

**Review by cpfergus1 — COMMENTED** [review]
Looks good, just have a question on how find_or_create_migration_content_type is intended to work

File: lib/contentful_migrations/migration_content_type.rb (hunk 2 above)
Comment (paulstraw) [reply lib/contentful_migrations/migration_content_type.rb:25]: Confirmed that this does work as expected IRL outside of a test environment!

**Comment by paulstraw** [comment]
@cpfergus1 other than the sidebar woes (now documented in Notion + ENG-2445), this seems to work great for the existing migrations. I've got a local changeset of the workaround for ENG-2445 on the Cometeer app, and will PR that with the Gemfile.lock update for this once it's merged! We don't currently have reviews required on this repo, but mind giving it a final check/approval?

**Review by cpfergus1 — APPROVED** [review]
Thank you so much!

**Pull request merged**